include/spdlog/sinks/rotating_file_sink-inl.h

```diff
@@ -41,14 +42,28 @@ SPDLOG_INLINE rotating_file_sink<Mutex>::rotating_file_sink(
     if (max_files > 200000) {
         throw_spdlog_ex("rotating sink constructor: max_files arg cannot exceed 200000");
     }
-    file_helper_.open(calc_filename(base_filename_, 0));
+    file_helper_.open(get_filename_for_rotation_(base_filename_, 0));
     current_size_ = file_helper_.size();  // expensive. called only once
     if (rotate_on_open && current_size_ > 0) {
         rotate_();
         current_size_ = 0;
     }
 }
 
+template <typename Mutex>
+SPDLOG_INLINE void rotating_file_sink<Mutex>::set_rotate_filename_format(std::function<filename_t(const filename_t &filename, std::size_t index)> rotation_file_format) {
```

Comment:
I don`t think `set_rotate_filename_format()` is necessary because it causes bugs.
It is not thread-safe and if the filename pattern changes during rotation, the rotation will not work.

`rotation_file_format_` should only be set in the constructor.

Reply:
Sure, updated.